src/zkstar.rs

```diff
@@ -0,0 +1,118 @@
+use crate::errors::InternalError;
+use libpaillier::unknown_order::BigNumber;
+use serde::{Deserialize, Serialize};
+
+/// A warpper around the modulus of the group [ZStarN] is defined over
+/// This allows passing the modulus by reference and avoid serializing it
+#[derive(Serialize)]
+struct ZStarNBuilder {
+    N: BigNumber,
+}
+
+impl ZStarNBuilder {
+    pub fn new(n: BigNumber) -> Self {
```

Comment:
thought: When we integrate this into the library, we might find that we are usually making this from either a `PaillierEncryptionKey` or a `RingPedersen`. We might restrict this API if that's the case (like a `from_paillier_key` and `from_ring_pedersen` method instead of generic, public `new`), but I'm not sure yet.

Reply:
I am reading this as: we will decide this during a second PR when we go ahead and start replacing all the `BigNumber` in the proofs and see what the library actually requires :)

Reply:
Exactly.